src/ansi-c/c_typecheck_code.cpp

```diff
@@ -869,80 +888,121 @@ void c_typecheck_baset::typecheck_spec_assigns_condition(exprt &condition)
       << eom;
   }
 
-  // fatal errors
-  bool must_throw = false;
-
   if(condition.type().id() == ID_empty)
   {
-    must_throw = true;
-    error().source_location = condition.source_location();
-    error() << "void-typed expressions not allowed as assigns clause conditions"
-            << eom;
+    std::ostringstream error_message;
+    error_message
+      << "void-typed expressions not allowed as assigns clause conditions";
+    throw invalid_source_file_exceptiont{
+      error_message.str(), condition.source_location()};
   }
 
   if(has_subexpr(condition, [&](const exprt &subexpr) {
        return can_cast_expr<side_effect_exprt>(subexpr) &&
               !can_cast_expr<side_effect_expr_function_callt>(subexpr);
      }))
   {
-    must_throw = true;
-    error().source_location = condition.source_location();
-    error() << "side-effects not allowed in assigns clause conditions" << eom;
+    std::ostringstream error_message;
+    error_message << "side-effects not allowed in assigns clause conditions";
+    throw invalid_source_file_exceptiont{
+      error_message.str(), condition.source_location()};
   }
 
   if(has_subexpr(condition, ID_if))
   {
-    must_throw = true;
-    error().source_location = condition.source_location();
-    error() << "ternary expressions not allowed in assigns "
-               "clause conditions"
-            << eom;
+    std::ostringstream error_message;
+    error_message
+      << "ternary expressions not allowed in assigns clause conditions";
+    throw invalid_source_file_exceptiont{
+      error_message.str(), condition.source_location()};
   }
-
-  if(must_throw)
-    throw 0;
 }
 
-void c_typecheck_baset::typecheck_spec_assigns_target(exprt &target)
+void c_typecheck_baset::typecheck_conditional_targets(
+  exprt::operandst &targets,
+  const std::function<void(exprt &)> typecheck_target,
+  const std::string &clause_type)
 {
-  // compute type
-  typecheck_expr(target);
+  exprt::operandst tmp;
```

Comment:
Although we do it way too many times across the code base: could you try to find a more descriptive name than "tmp?"

Reply:
is `new_targets` ok ?